src/connection.c

```diff
@@ -303,11 +303,14 @@ struct connection *conn_get_raw_server(struct context *ctx)
     int i;
     struct connection *server = NULL;
 
-    for (i = 0; i < config.node.len; i++) {
-        server = conn_get_server_from_pool(ctx, &config.node.addr[i], false);
+    struct node_conf *node = config.node;
+    conf_node_inc_ref(node);
```

Comment:
Note that there is race condition here. After you get `node`, other threads may call `conf_node_dec_ref` and destroy it which will cause corruption in `conf_node_inc_ref(node)`. We do need lock here and this is not likely to bring notable performance problem since the critical section here is so small.